GVFS/GVFS.UnitTests/Common/PathsTests.cs

```diff
@@ -31,5 +36,56 @@ public void CanConvertOSPathToGitFormat()
             string doubleTransformedPath = Paths.ConvertPathToGitFormat(actualTransformedPath);
             doubleTransformedPath.ShouldEqual(expectedGitPath);
         }
+
+        [TestCase(@"C:\", @"C:\a\b\c\file.txt", @"a\b\c\file.txt")]
+        [TestCase(@"C:\a\", @"C:\a\b\c\file.txt", @"b\c\file.txt")]
+        [TestCase(@"C:\a\b\", @"C:\a\b\c\file.txt", @"c\file.txt")]
+        [TestCase(@"C:\a\b\c\", @"C:\a\b\c\file.txt", @"file.txt")]
+        [TestCase(@"C:\a\d\e\", @"C:\a\b\c\file.txt", @"..\..\b\c\file.txt")]
+        [TestCase(@"C:\a\b\..\b\d\", @"C:\a\b\c\..\c\file.txt", @"..\c\file.txt")]
+        [TestCase(@"C:\a\", @"C:\a\b\c\", @"b\c\")]
+        [TestCase(@"C:\a\b\c\file.txt", @"C:\a\b\c\file.txt", @".")]
+        [TestCase(@"C:\a\b\c\", @"C:\a\b\c\", @".")]
+        [TestCase(@"Z:\a\b\", @"C:\a\b\c\file.txt", @"C:\a\b\c\file.txt")]
+        [TestCase(@"C:\a\spacey dir\", @"C:\a\spacey dir\c\spacey file.txt", @"c\spacey file.txt")]
+        public void GetRelativePath_Windows(string relativeTo, string path, string expected)
+        {
+            var actual = Paths.GetRelativePath(relativeTo, path);
+
+            // If we're running on non-Windows the relative path is returned with platform-
+            // native path separators, even if a Windows-style path was passed in.
+            // Convert it back to the Windows-style separator "\".
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))

```

Comment:
I don't think we should make tests behave differently based on what platform they're running on. What is the actual goal here?